salt/minion.py

```diff
@@ -3641,6 +3691,8 @@ def _process_event(self, raw):
         # TODO: cleanup: Move down into event class
         mtag, data = self.local.event.unpack(raw)
         log.trace("Got event %s", mtag)  # pylint: disable=no-member
+        if 'tgt_return' in mtag:
+            data["syndic_id"] = self.opts["id"]
```

Comment:
I'm not sure if there's a better way to get the ID of the syndic that's returning this data to the MoM.

Reply:
After playing around a little, pretty sure the answer is "no". The only place that the syndic/minion id will be present is in the syndic service itself. Without publishing an event per syndic (ew, gross), basically the only way is to inject the data into the event stream when we go to forward it onto the MoM.

I *think* I'm actually seeing the event from the master as *well* as the forwarded syndic event show up on the MoM, which is uh... weird. But it doesn't seem to have a huge impact.